tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -143,6 +145,139 @@ def test_fail_on_multiple_drivers():
         extract_db_type_from_uri("mysql+pymsql+pyodbc://...")
 
 
+@dataclass
+class TestDatabase:
+    db_url: str
+    store: SqlAlchemyStore
+
+
+@pytest.fixture
+def test_database() -> TestDatabase:
+    temp_dbfile, db_url = _setup_db_uri()
+    store = _get_store(db_url)
+    yield TestDatabase(db_url, store)
+    _cleanup_database(temp_dbfile, store)
+
+
+def _get_store(db_uri=""):
+    return SqlAlchemyStore(db_uri, ARTIFACT_URI)
+
+
+def _setup_db_uri():
+    if uri := MLFLOW_TRACKING_URI.get():
+        temp_dbfile = None
+        db_url = uri
+    else:
+        fd, temp_dbfile = tempfile.mkstemp()
+        # Close handle immediately so that we can remove the file later on in Windows
+        os.close(fd)
+        db_url = f"{DB_URI}{temp_dbfile}"
+    return temp_dbfile, db_url
```

Comment:
/ai propose a fix using pytest's tmp_path fixture.

Reply:
/ai doesn't look bad. `temp_dbfile` and `os.close(fd)` can be removed.

Reply:
Looks good. Can we apply this change?

Reply:
Updated in https://github.com/mlflow/mlflow/pull/10103/commits/8e0e366d381b14256ccc3ce04c9faeebd08ed5bf